Sources/Queues/QueueWorker.swift

```diff
@@ -15,19 +15,23 @@ public struct QueueWorker: Sendable {
     /// Actually run the queue. This is a thin wrapper for ELF-style callers.
     public func run() -> EventLoopFuture<Void> {
         self.queue.eventLoop.makeFutureWithTask {
-            try await self.run()
+            var hasWork = true
+            while hasWork {
+                hasWork = try await self.run()
+            }
         }
     }
     
     /// Pop a job off the queue and try to run it. If no jobs are available, do nothing.
-    public func run() async throws {
+    public func run() async throws -> Bool {
```

Comment:
This is a breaking change - we'll need to introduce a new function and deprecate this one

Reply:
Yep, I think I had also somewhat misunderstood the point of this API too. My hopefully corrected understanding is that `run() async` and `run() -> EventLoopFuture<Void>` are the two entrypoints to the worker, the former for async environments and the latter for non-async. As far as I can tell this package itself only uses the latter in the queue command, but that both are package public API.

Based on this, I've reverted the change and instead now have a private `runOneJob() -> Bool`, and both of the public entrypoints now have the run-until-done behaviour.